### PR TITLE
Change thermal bc interface and introduce enums for BC types

### LINKED ISSUE #52: Need to find a better way to deal with thermal BC

### PR DESCRIPTION
**Pull request by tduretz**

### CODE REVIEW

File: MDLIB/Main_DOODZ.c [hunk 1]
```diff
@@ -351,7 +353,7 @@ void RunMDOODZ(char *inputFileName, MdoodzSetup *setup) {
         Initialise1DArrayDouble( mesh.szzd,  (mesh.Nx-1)*(mesh.Nz-1), 0.0 );
         Initialise1DArrayDouble( mesh.sxz,   (mesh.Nx)  *(mesh.Nz)  , 0.0 );
         // Generate deformation maps
-        if (input.model.def_maps == 1 ) GenerateDeformationMaps( &mesh, &input.materials, &input.model, Nmodel, &input.scaling );
+        if (input.model.deformation_maps == 1 ) GenerateDeformationMaps( &mesh, &input.materials, &input.model, Nmodel, &input.scaling );
```
Comment (kulakovri) [inline MDLIB/Main_DOODZ.c:363]: What are deformation maps and when we should use them?

File: SETS/CollisionZone.c [hunk 2]
```diff
@@ -48,35 +48,20 @@ double SetDensity(MdoodzInput *input, Coordinates coordinates, int phase) {
...
-    bc.value = surfaceTemperature;
-    bc.type  = 1;
-  } else {
-    bc.value = 0.0;
-    bc.type  = 0;
+  SetBC     bc;
+  double surface_temperature =          zeroC  / instance->scaling.T;
+  double mantle_temperature  = (1330. + zeroC) / instance->scaling.T;
+  if (position == S) {
+    bc.type  = constant_temperature;
```
Comment (kulakovri) [inline SETS/CollisionZone.c:55]: looks very clear now!

File: MDLIB/include/mdoodz.h [hunk 3]
```diff
@@ -212,11 +212,18 @@ typedef enum {
...
+    constant_heatflux      = 0,
+    constant_temperature   = 1,
+    periodic_temeprature_x =-2,
 } Thermal_BC;
 
+// Chemical boundary condition types
+typedef enum {
+    constant_Xflux    = 0,
+    constant_X        = 1,
+    periodic_X_x      =-2,
```
Comment (kulakovri) [inline MDLIB/include/mdoodz.h]: what is periodic_X_x? More specifically, what does the second x stadnds for?

Comment (kulakovri) [reply MDLIB/include/mdoodz.h]: It can only be applicable to x dimension?

File: SETS/ChemicalDiffusion.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  const double T_init = (input->model.user0 + zeroC) / input->scaling.T;
+  const double P_init = (input->model.bkg_pressure         ) / input->scaling.S;
+  if (1 == 0) {
+    return input->materials.rho[phase] * exp(input->materials.bet[phase]*P_init -  input->materials.alp[phase] * T_init);
+  } else {
+    return input->materials.rho[phase];
+  }
+}
+
+SetBC SetBCC(MdoodzInput *instance, POSITION position, double particleTemperature) {
```
Comment (kulakovri) [inline SETS/ChemicalDiffusion.c:60]: Will we need particle temperature here at all?

File: SETS/README.md [hunk 5]
```diff
@@ -1,49 +1,61 @@
 # Parameters and parameter values read from input text file:
 
 ##  Simulation start/restart from breakpoint files
-- `irestart`: 0 if simulation starts from beginning, 1 if simulation restarts from a step. Default: 00000
-- `istep`: last step written, indicates step to restart from if irestart is 1. Default: 0
+| Parameter  | Description                                                                 | Default value | Allowed values |
```
Comment (kulakovri) [inline SETS/README.md:4]: I did not notice that new file, but it looks great. Would be also nice to add a link to the main Readme

File: MDLIB/Main_DOODZ.c (hunk 1 above)
Comment (tduretz) [reply MDLIB/Main_DOODZ.c:363]: we use them to generate maps of deformation mechanisms, mostly to compare to published data. 
You can have a look at the figures there: https://academic.oup.com/gji/article-abstract/221/3/1605/5743420

File: SETS/CollisionZone.c (hunk 2 above)
Comment (tduretz) [reply SETS/CollisionZone.c:55]: :D

File: MDLIB/include/mdoodz.h (hunk 3 above)
Comment (tduretz) [reply MDLIB/include/mdoodz.h]: X is compositional variable and x is the axis

File: SETS/ChemicalDiffusion.c (hunk 4 above)
Comment (tduretz) [reply SETS/ChemicalDiffusion.c:60]: yes, potentially

File: SETS/README.md (hunk 5 above)
Comment (tduretz) [reply SETS/README.md:4]: if you or @halterw  knows how to do so, it will be welcome!

**Pull request merged**